src/java.base/share/classes/sun/nio/ch/SocketAdaptor.java

```diff
@@ -85,6 +86,14 @@ public void connect(SocketAddress remote) throws IOException {
     public void connect(SocketAddress remote, int timeout) throws IOException {
         if (remote == null)
             throw new IllegalArgumentException("connect: The address can't be null");
+        if (remote instanceof InetSocketAddress isa && isa.isUnresolved()) {
+            if (!sc.isOpen())
+                    throw new SocketException("Socket is closed");
+            if (sc.isConnected())
+                    throw new SocketException("Already connected");
```

Comment:
Would you mind changing this to use 4-space instead of 8-space indent?

Reply:
Fixed in aaf398f328cfb4771df1a7af22090c5117ec6055.